pvlib/singlediode.py

```diff
@@ -247,22 +249,29 @@ def bishop88_i_from_v(voltage, photocurrent, saturation_current,
     method : str, default 'newton'
        Either ``'newton'`` or ``'brentq'``. ''method'' must be ``'newton'``
        if ``breakdown_factor`` is not 0.
+    method_kwargs : dict
```

Comment:
Good point. It looks a bit weird how `kwargs` are documented. Example provided from [`numpydoc`](https://numpydoc.readthedocs.io/en/latest/format.html#parameters):
```
**kwargs : dict, optional
    Extra arguments to `metric`: refer to each metric documentation for a
    list of all possible arguments.
```
However, in PVLIB (taken from [pvlib.location.Location.get_clearsky](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.location.Location.get_clearsky.html)) it is usually
```
kwargs
            Extra parameters passed to the relevant functions. Climatological
            values are assumed in many cases. See source code for details!
```
[This stackoverflow post](https://stackoverflow.com/questions/62511086/how-to-document-kwargs-according-to-numpy-style-docstring) does give some more insight that could be useful.

I like the function signature way of passing parameters, given they are well documented, and document kwargs with `**kwargs` without type definition.
If no objections are made, I'll add some use examples both for `kwargs` and `full_output=True`.

Reply:
> I like the function signature way of passing parameters, given they are well documented, and document kwargs with `**kwargs` without type definition.

I didn't justify my opinion. Right now, `**kwargs` is well spread in PVLIB (well, in docs it usually is `kwargs`, without preceding `**`) to pass parameters to other functions, so I feel the dict variant a bit disruptive in that sense. I added the asterisks cause I think they hold an important meaning. Of course I don't have a strong opinion and if you tell me otherwise I'll change it without problems.

Something that puzzles me a bit is using `full_output` inside the functions, making it not just a keyword arg passed to the root finder. However, IMO this is a right approach for now. Adding a parameter for that seems redundant.

Reply:
I thought there might be others with opinions about this...  

There's a similar situation in scipy least-squares.  I think having the extra arguments that will be passed to another function reside inside a dictionary (or and args list, like newton does) clearly separates them from those that are used within.

Reply:
Okay, I've changed the behaviour although that still feels like the pattern for a user-provided function.

tests (with few unrelated exceptions I believe), doctest examples are passing and all seems good to me. Do you mind having a look specially at the formatting of docs?

Reply:
I guess choosing the method name is an indirect way of giving a user-provided function.  :)
I'm not that good at the doc formatting so I'll defer to others.